### ISSUE TITLE
event handler can't connect to postgres port on 5432

### ISSUE THREAD

**Issue by anasfik** [UTXOnly/nost-py#45]
Hello there,

I cloned the relay into an empty VM running Ubuntu 22.04, and set up my `.env` file:

```
PGDATABASE=nostr                                                                                                                                                                                                  
PGUSER=nostr                                                                                                                                                                                                      
PGPASSWORD=nostr                                                                                                                                                                                                  
PGPORT=5432                                                                                                                                                                                                       
PGHOST=localhost                                                                                                                                                                                                  
DD_ROLE_PASSWORD=                                                                                                                                                                                                 
DD_ROLE_NAME=                                                                                                                                                                                                     
DD_ENV=                                                                                                                                                                                                           
DD_HOSTNAME=                                                                                                                                                                                                      
DD_API_KEY=                                                                                                                                                                                                       
DOMAIN_NAME=relay.tribly.co                                                                                                                                                                                       
HEX_PUBKEY=<YOUR_HEX_PUBKEY>                                                                                                                                                                                      
CONTACT=your_email@email.com                                                                                                                                                                                      
ENV_FILE_PATH=./docker_stuff/.env                                                                                                                                                                                 
NGINX_FILE_PATH=/etc/nginx/sites-available/default                                                                                                                                                                
VERSION=v0.8                                                                                                                                                                                                                                                                                                                                                                                                                                                                
```

then runned the menu script with `python3 menu.py`, runned the `Execute server setup script` command, it goes very well, then I runned the `Start Nostpy relay`, provided my encryption password and the containers seems to work fine, so I did some more configuration to reverse proxy the relay port (8008) with nginx since we need to set it in another nginx config file other than `default`, set a TLS certificate and so the relay seems to connect fine at `wss://relay.tribly.co`.

However, after trying to use it with a client, it fails to accept sent events (it don't return an `OK` command), so I checked the docker containers logs with `docker-compose logs`:

```
datadog-agent        | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
datadog-agent        | [s6-init] ensuring user provided files have correct perms...exited 0.
datadog-agent        | [fix-attrs.d] applying ownership & permissions fixes...
datadog-agent        | [fix-attrs.d] done.
datadog-agent        | [cont-init.d] executing container initialization scripts...
datadog-agent        | [cont-init.d] 01-check-apikey.sh: executing...
datadog-agent        |
datadog-agent        | ==================================================================================
datadog-agent        | You must set an DD_API_KEY environment variable to run the Datadog Agent container
datadog-agent        | ==================================================================================
datadog-agent        |
datadog-agent        | [cont-init.d] 01-check-apikey.sh: exited 1.
datadog-agent        | [cont-finish.d] executing container finish scripts...
datadog-agent        | [cont-finish.d] done.
datadog-agent        | [s6-finish] waiting for services.
datadog-agent        | [s6-finish] sending all processes the TERM signal.
datadog-agent        | [s6-finish] sending all processes the KILL signal and exiting.
postgres_1           |
postgres_1           | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres_1           |
postgres_1           | 2024-02-27 22:52:38.482 UTC [1] LOG:  starting PostgreSQL 14.11 (Debian 14.11-1.pgdg120+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
postgres_1           | 2024-02-27 22:52:38.483 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1           | 2024-02-27 22:52:38.483 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1           | 2024-02-27 22:52:38.488 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1           | 2024-02-27 22:52:38.497 UTC [28] LOG:  database system was shut down at 2024-02-27 22:50:28 UTC
postgres_1           | 2024-02-27 22:52:38.517 UTC [1] LOG:  database system is ready to accept connections
event_handler_1      | failed to send, dropping 2 traces to intake at http://datadog-agent:8126/v0.5/traces after 3 retries
event_handler_1      | INFO:     Started server process [1]
event_handler_1      | INFO:     Waiting for application startup.
event_handler_1      | INFO:     Application startup complete.
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | INFO:     Uvicorn running on http://0.0.0.0:80 (Press CTRL+C to quit)
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 1 additional messages skipped
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Unable to export profile: ddtrace.profiling.exporter.ExportError: HTTP upload request failed: [Errno -3] Temporary failure in name resolution. Ignoring.
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 9 additional messages skipped
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Unable to export profile: ddtrace.profiling.exporter.ExportError: HTTP upload request failed: [Errno -3] Temporary failure in name resolution. Ignoring.
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 11 additional messages skipped
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Unable to export profile: ddtrace.profiling.exporter.ExportError: HTTP upload request failed: [Errno -3] Temporary failure in name resolution. Ignoring.
event_handler_1      | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 11 additional messages skipped
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Unable to export profile: ddtrace.profiling.exporter.ExportError: HTTP upload request failed: [Errno -3] Temporary failure in name resolution. Ignoring.
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 11 additional messages skipped
event_handler_1      | error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | INFO:     172.28.0.2:52276 - "POST /new_event HTTP/1.1" 500 Internal Server Error
event_handler_1      | ERROR:    Exception in ASGI application
event_handler_1      | Traceback (most recent call last):
event_handler_1      |   File "/app/event_handler.py", line 114, in handle_new_event
event_handler_1      |     async with request.app.async_pool.connection() as conn:
event_handler_1      |   File "/usr/local/lib/python3.9/contextlib.py", line 181, in __aenter__
event_handler_1      |     return await self.gen.__anext__()
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/psycopg_pool/pool_async.py", line 190, in connection
event_handler_1      |     conn = await self.getconn(timeout=timeout)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/psycopg_pool/pool_async.py", line 225, in getconn
event_handler_1      |     raise PoolTimeout(
event_handler_1      | psycopg_pool.PoolTimeout: couldn't get a connection after 30.00 sec
event_handler_1      |
event_handler_1      | During handling of the above exception, another exception occurred:
event_handler_1      |
event_handler_1      | Traceback (most recent call last):
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
event_handler_1      |     result = await app(  # type: ignore[func-returns-value]
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
event_handler_1      |     return await self.app(scope, receive, send)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 292, in __call__
event_handler_1      |     await super().__call__(scope, receive, send)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/starlette/applications.py", line 122, in __call__
event_handler_1      |     await self.middleware_stack(scope, receive, send)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/ddtrace/contrib/asgi/middleware.py", line 270, in __call__
event_handler_1      |     reraise(exc_type, exc_val, exc_tb)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/six.py", line 719, in reraise
event_handler_1      |     raise value
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/ddtrace/contrib/asgi/middleware.py", line 265, in __call__
event_handler_1      |     return await self.app(scope, receive, wrapped_send)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
event_handler_1      |     raise exc
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
event_handler_1      |     await self.app(scope, receive, _send)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
event_handler_1      |     raise exc
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
event_handler_1      |     await self.app(scope, receive, sender)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
event_handler_1      |     raise e
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
event_handler_1      |     await self.app(scope, receive, send)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 718, in __call__
event_handler_1      |     await route.handle(scope, receive, send)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 276, in handle
event_handler_1      |     await self.app(scope, receive, send)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 66, in app
event_handler_1      |     response = await func(request)
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 273, in app
event_handler_1      |     raw_response = await run_endpoint_function(
event_handler_1      |   File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 190, in run_endpoint_function
event_handler_1      |     return await dependant.call(**values)
event_handler_1      |   File "/app/event_handler.py", line 139, in handle_new_event
event_handler_1      |     conn.rollback()
event_handler_1      | UnboundLocalError: local variable 'conn' referenced before assignment
event_handler_1      | failed to send, dropping 1 traces to intake at http://172.28.0.5:8126/v0.5/traces after 3 retries
websocket_handler_1  | failed to send, dropping 2 traces to intake at http://datadog-agent:8126/v0.5/traces after 3 retries
websocket_handler_1  | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket
websocket_handler_1  | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 4 additional messages skipped
websocket_handler_1  | Unable to export profile: ddtrace.profiling.exporter.ExportError: HTTP upload request failed: [Errno -3] Temporary failure in name resolution. Ignoring.
websocket_handler_1  | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 17 additional messages skipped
websocket_handler_1  | Unable to export profile: ddtrace.profiling.exporter.ExportError: HTTP upload request failed: [Errno -3] Temporary failure in name resolution. Ignoring.
websocket_handler_1  | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 17 additional messages skipped
websocket_handler_1  | Unable to export profile: ddtrace.profiling.exporter.ExportError: HTTP upload request failed: [Errno -3] Temporary failure in name resolution. Ignoring.
websocket_handler_1  | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 17 additional messages skipped
websocket_handler_1  | Unable to export profile: ddtrace.profiling.exporter.ExportError: HTTP upload request failed: [Errno -3] Temporary failure in name resolution. Ignoring.
websocket_handler_1  | Error submitting packet: [Errno -3] Temporary failure in name resolution, dropping the packet and closing the socket, 17 additional messages skipped
websocket_handler_1  | failed to send, dropping 1 traces to intake at http://172.28.0.5:8126/v0.5/traces after 3 retries
redis-cache          | 1:C 27 Feb 2024 22:52:38.227 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis-cache          | 1:C 27 Feb 2024 22:52:38.227 * oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis-cache          | 1:C 27 Feb 2024 22:52:38.227 * Redis version=7.2.4, bits=64, commit=00000000, modified=0, pid=1, just started
redis-cache          | 1:C 27 Feb 2024 22:52:38.227 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
redis-cache          | 1:M 27 Feb 2024 22:52:38.228 * monotonic clock: POSIX clock_gettime
redis-cache          | 1:M 27 Feb 2024 22:52:38.228 * Running mode=standalone, port=6379.
redis-cache          | 1:M 27 Feb 2024 22:52:38.229 * Server initialized
redis-cache          | 1:M 27 Feb 2024 22:52:38.229 * Ready to accept connections tcp
```

I kinda know what probably causes this, which is that the `event_handler_1` can't connect to Postgres via port 5432:

```
 error connecting in 'pool-1': connection is bad: :1", port 5432 failed: Cannot assign requested address
event_handler_1      |  Is the server running on that host and accepting TCP/IP connections?
event_handler_1      | E
```

I suspected then that Postgres might not be running on that port so I ran `netstat -tulnp`:
```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:5432            0.0.0.0:*               LISTEN      553733/docker-proxy
tcp        0      0 127.0.0.1:46605         0.0.0.0:*               LISTEN      2230/containerd
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      639/systemd-resolve
tcp        0      0 0.0.0.0:8008            0.0.0.0:*               LISTEN      553707/docker-proxy
tcp        0      0 0.0.0.0:6379            0.0.0.0:*               LISTEN      553769/docker-proxy
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      778/sshd: /usr/sbin
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      3803/nginx: master
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      3803/nginx: master
tcp6       0      0 :::5432                 :::*                    LISTEN      553738/docker-proxy
tcp6       0      0 :::8008                 :::*                    LISTEN      553714/docker-proxy
tcp6       0      0 :::6379                 :::*                    LISTEN      553776/docker-proxy
tcp6       0      0 :::22                   :::*                    LISTEN      778/sshd: /usr/sbin
tcp6       0      0 :::3001                 :::*                    LISTEN      1332/node
udp        0      0 127.0.0.53:53           0.0.0.0:*                           639/systemd-resolve
```

and it seems like Docker is listening on that port.

do you have any helpful thoughts about it, please let me know if I should provide any more information or context about it.

Thank you!

**Comment by UTXOnly** [comment]
Hey @anasfik thanks for bringing this to my attention!

I was able to recreate the issue and apologies for not catching this earlier. The issue is due to the `event_handler` service trying to submit data to `localhost` which would be relative to the container the `event_handler` service is running instead of the `localhost` for the VM the compose stack is running on. 

Could you please try changing the below line in your `.env` file.

```
PGHOST=172.28.0.4
```

This was an oversight in the documentation in the README, which did not match the actual `.env` file in the repo, which has  the correct IP address for the `postgres` container.

I have addressed this issue in PR #46.

Please let me know if this resolves your issue or feel free to reach out with any additional questions/concerns.

Thanks!

**Comment by UTXOnly** [comment]
Closing as confirmed relay started on issue #53
